spec/views/account/register.html.erb_spec.rb

```diff
@@ -29,25 +29,78 @@
 require 'spec_helper'
 
 describe 'account/register', type: :view do
-  before do
-    assign(:user, user)
-    render
+  let(:user) { FactoryGirl.build :user, auth_source: nil }
+
+  context 'with the email_login setting disabled (default value)' do
+    before do
+      Setting.email_login = 0
```

Comment:
Setting the values explicitly can have side effects with other tests.

It is therefore advisable to mock the value instead.